base/abstractarray.jl

```diff
@@ -113,10 +113,12 @@ linearindexing{T<:AbstractArray}(::Type{T}) = LinearSlow()
 linearindexing{T<:Array}(::Type{T}) = LinearFast()
 linearindexing{T<:Range}(::Type{T}) = LinearFast()
 
-*(::LinearFast, ::LinearFast) = LinearFast()
-*(::LinearSlow, ::LinearFast) = LinearSlow()
-*(::LinearFast, ::LinearSlow) = LinearSlow()
-*(::LinearSlow, ::LinearSlow) = LinearSlow()
+linearindexing(A::AbstractArray, B::AbstractArray) = linearindexing(linearindexing(A), linearindexing(B))
+linearindexing(A::AbstractArray, B::AbstractArray...) = linearindexing(linearindexing(A), linearindexing(B...))
+linearindexing(::LinearFast, ::LinearFast) = LinearFast()
```

Comment:
Not if I understand this correctly...
If I'm right, I think @timholy needs to add a comment explaining why that is all spelled out that way, to avoid some later well meaning julian from optimizing away his optimization...
If you have A, B, C, D, and A, B, C are LinearFast, and D is LinearSlow, the way he has it does two fast operations, and one slow, but yours does 3 slow...